Upstream-Status: Backport

---
 configure.ac | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/configure.ac b/configure.ac
index 5430021..749d68a 100644
--- a/configure.ac
+++ b/configure.ac
@@ -937,6 +937,8 @@ cat > conftest.c <<EOF
         x86_64-linux-gnu
 # elif defined(__x86_64__) && defined(__ILP32__)
         x86_64-linux-gnux32
+# elif defined(__loongarch64)
+        loongarch64-linux-gnu
 # elif defined(__i386__)
         i386-linux-gnu
 # elif defined(__aarch64__) && defined(__AARCH64EL__)
